### PR TITLE
[InLong-4227][Sort] Sort lightwieght support extract data to ClickHouse

### LINKED ISSUE #4227: [Feature][Sort] Sort lightwieght support extract data to clickhouse
### Description

Sort lightwieght support extract data from pulsar

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thexiay**
### Title Name: [INLONG-4227][Sort] Sort lightwieght support extract data to clickhouse

Fixes #4227
### Motivation

Sort lightwieght support extract data to clickhouse

### Modifications

Add clickhouse load node.
Add jdbc-inlong table factory to enhance custom jdbc dialect.
Add Clickhouse jdbcdialect.

### CODE REVIEW

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/clickhouse/table/ClickHouseDialect.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.Optional;
+
+public class ClickHouseDialect extends AbstractJdbcDialect {
+
+    // Define MAX/MIN precision of TIMESTAMP type according to Mysql docs:
+    // https://clickhouse.com/docs/zh/sql-reference/data-types/datetime64
+    private static final int MAX_TIMESTAMP_PRECISION = 8;
+    private static final int MIN_TIMESTAMP_PRECISION = 0;
+
+    // Define MAX/MIN precision of DECIMAL type according to Mysql docs:
```
Comment (thexiay) [inline inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/clickhouse/table/ClickHouseDialect.java:37]: `Mysql` change to `ClcikHouse`

File: inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/ClickHouseSqlParserTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ClickHouseSqlParserTest {
+    public MySqlExtractNode buildMySQLExtractNode(String id) {
+        List<FieldInfo> fields = Arrays.asList(new FieldInfo("id", new LongFormatInfo()),
+                new FieldInfo("name", new StringFormatInfo()));
+        // if you hope hive load mode of append, please add this config
```
Comment (healchow) [inline inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/ClickHouseSqlParserTest.java:50]: Why use `hive` here?

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/jdbc/table/AbstractJdbcDialect.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.TimestampType;
+import org.apache.flink.table.types.logical.VarBinaryType;
+
+import java.util.List;
+
+public abstract class AbstractJdbcDialect implements JdbcDialect {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/jdbc/table/AbstractJdbcDialect.java:32]: Please add some java comments.

File: inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/jdbc/table/JdbcDialects.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.apache.flink.connector.jdbc.dialect.JdbcDialect;
+import org.apache.flink.connector.jdbc.dialect.MySQLDialect;
+import org.apache.flink.connector.jdbc.dialect.PostgresDialect;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** Default JDBC dialects. */
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/src/main/java/org/apache/inlong/sort/flink/jdbc/table/JdbcDialects.java:29]: Check if this comment is normative?

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/PostgresConstant.java [hunk 5]
```diff
@@ -49,6 +49,6 @@ public class PostgresConstant {
 
     public static final String URL = "url";
 
-    public static final String JDBC = "jdbc";
+    public static final String JDBC = "jdbc-inlong";
```
[the remark this answers is not in the document]
Comment (thexiay) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/PostgresConstant.java:52]: all jdbc connector both use jdbc-inlong conenctor

File: inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/ClickHouseSqlParserTest.java (hunk 2 above)
Comment (thexiay) [reply inlong-sort/sort-single-tenant/src/test/java/org/apache/inlong/sort/singletenant/flink/parser/ClickHouseSqlParserTest.java:50]: comment error

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/ClickHouseLoadNode.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Override
+    public String genTableName() {
+        return tableName;
+    }
+
+    @Override
+    public String getPrimaryKey() {
+        return super.getPrimaryKey();
```
[the remark this answers is not in the document]
Comment (thexiay) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/ClickHouseLoadNode.java:109]: clickhouse don's support upsert, jdbc relay on primary key to support upsert，so here we should not define clickhouse table a primary key.

[the remark this answers is not in the document]
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/ClickHouseLoadNode.java:108]: ok.

**Review by yunqingmoswu — APPROVED** [review]

**Comment by thexiay** [comment]
i upgrade flink-sql-connector-postgres-cdc version because 2.0.1 version contains kafka code conflict with kafka-client.It's Qualified class name is same.

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**